File: trunk/src/scala/org/codehaus/testdox/intellij/TestMethod.scala
package org.codehaus.testdox.intellij

import com.intellij.openapi.actionSystem.Presentation
import com.intellij.psi.PsiMethod
import javax.swing.Icon

class TestMethod(psiMethod: PsiMethod, editorApi: EditorApi, sentenceManager: SentenceManager) extends AbstractTestElement {

  def methodName: String = psiMethod.getName()

  override val psiElement = psiMethod

  override def jumpToPsiElement(): Boolean = editorApi.jumpToPsiElement(psiElement)

  override def displayString: String = sentenceManager.buildSentence(methodName)

  override def icon: Icon = Icons.getIcon(Icons.DOX_ICON)

  override def update(presentation: Presentation) = presentation.setEnabled(true)

  override def rename(controller: TestDoxController) = controller.startRename(this)

  override def delete(controller: TestDoxController) = editorApi.delete(psiElement)

  override def compare(that: TestElement): Int = {
    if (that.isInstanceOf[TestMethod])
      this.displayString.compareTo(that.displayString)
    else
      0
  }
}

object TestMethod {

  val EMPTY_ARRAY = new Array[TestMethod](0)
}

File: branches/pre-scala-migration/src/java/org/codehaus/testdox/intellij/TestMethod.java
package org.codehaus.testdox.intellij;

import javax.swing.Icon;

import com.intellij.openapi.actionSystem.Presentation;
import com.intellij.psi.PsiElement;
import com.intellij.psi.PsiMethod;

public class TestMethod extends AbstractTestElement {

    public static final TestMethod[] EMPTY_ARRAY = new TestMethod[0];

    private final PsiMethod method;
    private final EditorApi editorApi;
    private final SentenceManager sentenceManager;

    public TestMethod(PsiMethod method, EditorApi editorApi, SentenceManager sentenceManager) {
        this.method = method;
        this.editorApi = editorApi;
        this.sentenceManager = sentenceManager;
    }

    public String getMethodName() {
        return method.getName();
    }

    public PsiElement getPsiElement() {
        return method;
    }

    public boolean jumpToPsiElement() {
        return editorApi.jumpToPsiElement(getPsiElement());
    }

    public String getDisplayString() {
        return sentenceManager.buildSentence(getMethodName());
    }

    public Icon getIcon() {
        return IconHelper.getIcon(IconHelper.DOX_ICON);
    }

    public void updatePresentation(Presentation presentation) {
        presentation.setEnabled(true);
    }

    public void rename(TestDoxController testDoxController) {
        testDoxController.startRename(this);
    }

    public void delete(TestDoxController testDoxController) {
        editorApi.delete(getPsiElement());
    }

    public int compareTo(Object object) {
        if (object instanceof TestMethod) {
            return getDisplayString().compareTo(((TestMethod) object).getDisplayString());
        }
        return 0;
    }
}

